src/common/ram/cabin.go

```diff
@@ -0,0 +1,110 @@
+// Copyright Project Harbor Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ram
+
+import (
+	"fmt"
+
+	"github.com/casbin/casbin"
+	"github.com/casbin/casbin/model"
+	"github.com/casbin/casbin/persist"
+)
+
+var modelText = `
+# Request definition
+[request_definition]
+r = sub, obj, act
+
+# Policy definition
+[policy_definition]
+p = sub, obj, act
+
+# Role definition
+[role_definition]
+g = _, _
+
+# Policy effect
+[policy_effect]
+e = some(where (p.eft == allow))
+
+# Matchers
+[matchers]
+m = g(r.sub, p.sub) && r.obj == p.obj && (r.act == p.act || p.act == '*')
```

Comment:
Add more variation in obj matching.

Reply:
obj matching changed to keyMatch2 which suport * and :var